From e0548032ddef15ebf869a5f10182cc61738c6e4f Mon Sep 17 00:00:00 2001
From: Gaming4JC <g4jc@hyperbola.info>
Date: Sat, 14 May 2022 10:06:12 -0400
Subject: [PATCH 3/6] Hardcode AppName in nsAppRunner

Patches WM_ClassName and Application Name since we are keeping them unchanged for folder compatibility reasons.
Additional branding improvements of Help and Version String changes courtesy HarvettFox96.
---
 toolkit/xre/nsAppRunner.cpp | 14 +++++---------
 1 file changed, 5 insertions(+), 9 deletions(-)

diff --git a/toolkit/xre/nsAppRunner.cpp b/toolkit/xre/nsAppRunner.cpp
index 274631aac..1b47cf56a 100644
--- a/toolkit/xre/nsAppRunner.cpp
+++ b/toolkit/xre/nsAppRunner.cpp
@@ -1300,11 +1300,11 @@ DumpHelp()
 #endif
 #ifdef XP_UNIX
   printf("  --g-fatal-warnings                           Make all warnings fatal.\n"
-         "\n%s options\n", gAppData->name);
+         "\nIceweasel-UXP options\n", gAppData->name);
 #endif
 
   printf("  -h or --help                                 Print this message.\n"
-         "  -v or --version                              Print %s version.\n"
+         "  -v or --version                              Print Iceweasel-UXP version.\n"
          "  -P <profile>                                 Start with <profile>.\n"
          "  --profile <path>                             Start with profile at <path>.\n"
 #ifdef MC_BASILISK
@@ -1373,9 +1373,7 @@ static int MSCRTReportHook( int aReportType, char *aMessage, int *oReturnValue)
 static inline void
 DumpVersion()
 {
-  if (gAppData->vendor)
-    printf("%s ", gAppData->vendor);
-  printf("%s %s", gAppData->name, gAppData->version);
+  printf("Hyperbola Iceweasel-UXP %s", gAppData->version);
   if (gAppData->copyright)
       printf(", %s", gAppData->copyright);
   printf("\n");
@@ -3141,11 +3139,9 @@ XREMain::XRE_mainStartup(bool* aExitFlag)
     gdk_rgb_set_install(TRUE);
 #endif
 
-  // Set program name to the one defined in application.ini.
+  // Set WM_CLASS name
   {
-    nsAutoCString program(gAppData->name);
-    ToLowerCase(program);
-    g_set_prgname(program.get());
+    g_set_prgname("Iceweasel-UXP");
   }
 
   // Initialize GTK here for splash.
-- 
2.35.1

